examples/webbilling-demo/src/tests/tax-calculation.test.ts

```diff
@@ -0,0 +1,172 @@
+import { expect, test } from "@playwright/test";
+import {
+  enterCreditCardDetails,
+  enterEmailAndContinue,
+  getAllElementsByLocator,
+  getUserId,
+  setupTest,
+  startPurchaseFlow,
+} from "./utils.ts";
+
+const CARD_SELECTOR = "div.card";
+
+// Assuming there's a selector or some identifiable locator for the tax breakdown UI element
+const TAX_BREAKDOWN_ITEM_SELECTOR = ".rcb-pricing-table-row";
+const ALLOW_TAX_CALCULATION_FF =
+  process.env.VITE_ALLOW_TAX_CALCULATION_FF === "true";
+
+test.describe("Tax calculation breakdown", () => {
+  test.skip(
+    !ALLOW_TAX_CALCULATION_FF,
+    "Tax calculation breakdown is disabled. To enable, set VITE_ALLOW_TAX_CALCULATION_FF=true in the environment variables.",
+  );
+
+  test("should not display the tax breakdown at the enter email step", async ({
+    browser,
+    browserName,
+  }) => {
+    const userId = getUserId(browserName);
+
+    // Set up the page (standard user, location, default params)
+    const page = await setupTest(browser, userId);
+
+    // Select and make a purchase of the first available product card
+    const cards = await getAllElementsByLocator(page, CARD_SELECTOR);
+    const targetCard = cards[0];
+
+    // Perform a purchase action
+    await startPurchaseFlow(targetCard);
+
+    await expect(page.getByText("Total excluding tax")).not.toBeVisible();
+  });
+
+  test("should display the tax breakdown at the checkout step", async ({
```

Comment:
This will depend on the test runner location since the tax pre-calculation is based on the client IP.

Reply:
Yeah, this specific test can be flakey, I don't have a better idea on how to test the initial calculation though.